SCF-Session2-Question1/src/meta/helper/StaticCheckingAndCodeReviewQn1_HexCalcImpl.java

```diff
@@ -0,0 +1,356 @@
+			decimalNumber = decimalNumber/base;
+		}
+		return hexaDecimalString;
+	}
+	/*
+	 * (non-Javadoc)
+	 * @see meta.helper.HexCalc#convertToDecimal(java.lang.String, int)
+	 * @param number is a hexaDecimal value (letter are in Capital Format from  main method)
+	 * @param base is for base conversion value (in this program it is always 16)
+	 * @return A String which Contains Integer in form of String
+	 * Assumption number String Contains only capital characters
+	 * Assumption number String contains A valid HexNumber
+	 * Assumption number is  positive
+	 */
+	@Override
+	public String convertToDecimal( String number, int base ) {
+		String decimalValueString="";
+		int resultIntValue=0; // Store Converted Value From HexaDecimal to Decimal
+		int stringLength = number.length();
+		char hexCharArray[] = {'0','1','2','3','4','5','6','7','8','9','A','B','C','D','E','F'};
+		base = 16;
+		// this array is used for getting value of Hex Number
+		/*
+		 * this for loop is calculating the Decimal Value from HexaDecimal String
+		 * where we get each char from String and calculate corresponding it
+		 * and final value is calculated into resultIntValue
+		 * which converted to String and return 
+		 */
+		for(int i=0; i<stringLength;i++)
+		{
+			char currentChar = number.charAt(i);
+			int currentValue = (int) (getIndexOf(currentChar,hexCharArray)*Math.pow(base,stringLength-1-i));
+			// getIndexOf() function is calling for getting index of that value in array
+			resultIntValue += currentValue;
+		}
+		decimalValueString = ""+resultIntValue;
+		return decimalValueString;
+	}
+	/*
+	 * this is supporting function of ConvertToDecimal() function for getting index of character
+	 * @param currentChar is character of hex String
+	 * @param hexCharArray[] is array of HexaDecimal Values
+	 * @return index of that particular character
+	 * Assumption currentChar is a Valid Hex char
+	 */
+	public int getIndexOf(char currentChar, char[] hexCharArray)
+	{
+		int index=-1,noVal=0;
+		for(char ch: hexCharArray)
+		{
+			index++;
+			if(ch==currentChar)
+			{
+				return index;
+			}
+		}
+		/*
+		 * returns 0 if no character is found in the array
+		 * this means input hex value is wrong then
+		 * if we return -1 then it raise an error in that program so 
+		 * prevent from exception we retrun 0
+		 */
+		return noVal;
+	}
+	
+	/*
+	 * (non-Javadoc)
+	 * @see meta.helper.HexCalc#addTwoBaseNumbers(java.lang.String, java.lang.String, int)
+	 * @param firstNumber is a HexaDecimal String Value
+	 * @param secondNumber is a HexaDecimal String Value
+	 * @param base is for base conversion value (in this program it is always 16)
+	 * @return returns a Addition Of two hexaDeciaml Numbers in String form
+	 * Assumption String's Contains only capital characters
+	 * Assumption String's String contains A valid HexNumber
+	 * Assumption both values are positive
+	 */
+	@Override
+	public String addTwoBaseNumbers( String firstNumber, String secondNumber, int base ) {
```

Comment:
If we have to generalise the approach, we shall first convert number (in any base) to decimal, and then perform the arithmetic or logical operations. 
Here, again in every method, the code is repeated to convert it from hexadecimal to decimal.